alacritty_terminal/src/term/search.rs

```diff
@@ -881,19 +923,34 @@ mod tests {
         ");
 
         // Bottom to top.
-        let regex = RegexSearch::new("abc").unwrap();
+        let mut regex = RegexSearch::new("abc").unwrap();
         let start = Point::new(Line(1), Column(0));
         let end = Point::new(Line(0), Column(2));
         let match_start = Point::new(Line(0), Column(0));
         let match_end = Point::new(Line(0), Column(2));
-        assert_eq!(term.regex_search_right(&regex, start, end), Some(match_start..=match_end));
+        assert_eq!(term.regex_search_right(&mut regex, start, end), Some(match_start..=match_end));
 
         // Top to bottom.
-        let regex = RegexSearch::new("def").unwrap();
+        let mut regex = RegexSearch::new("def").unwrap();
         let start = Point::new(Line(0), Column(2));
         let end = Point::new(Line(1), Column(0));
         let match_start = Point::new(Line(1), Column(0));
         let match_end = Point::new(Line(1), Column(2));
-        assert_eq!(term.regex_search_left(&regex, start, end), Some(match_start..=match_end));
+        assert_eq!(term.regex_search_left(&mut regex, start, end), Some(match_start..=match_end));
+    }
+
+    #[test]
+    fn nfa_compile_error() {
+        assert!(RegexSearch::new("[0-9A-Za-z]{9999999}").is_err());
+    }
+
+    #[test]
+    fn runtime_cache_error() {
+        let term = mock_term(&str::repeat("i", 9999));
+
+        let mut regex = RegexSearch::new("[0-9A-Za-z]{9999}").unwrap();
+        let start = Point::new(Line(0), Column(0));
+        let end = Point::new(Line(0), Column(9999));
+        assert_eq!(term.regex_search_right(&mut regex, start, end), None);
```

Comment:
Would it be worth making these errors a little more descriptive, since they could in theory be hit by a crazy user configuration. Always nice to help tame insanity imo.